### PR TITLE
Extended brion::Synapse to also support non-merged synapse files

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by hernando** [comment]
Unit test missing

**Comment by eile** [comment]
No data to test against:
```
[shichi Equalizer master]% find ../BBPTestData/ -name nrn.h5.\*
../BBPTestData//ballAndStick/example.out/ncsStructural/nrn.h5.0
[shichi Equalizer master]%
```

**Comment by tribal-tec** [comment]
Then there should be test data added

File: doc/Changelog.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 Changelog {#Changelog}
 =========
 
+## git master {master)
+
+* Add brion::Target::parse() to resolve a given target name
```
Comment (tribal-tec) [inline doc/Changelog.md]: link to PR?

Comment (eile) [inline doc/Changelog.md]: Never been done here :P

Will do, but in general annoying: git push, open pr, edit changelog, git commit amend, git force push, retest this please...

**Comment by eile** [comment]
Updated - should address all comments.

File: tests/synapse.cpp [hunk 2]
```diff
@@ -247,3 +247,34 @@ BOOST_AUTO_TEST_CASE( test_perf )
...
+
+BOOST_AUTO_TEST_CASE( read_unmerged )
+{
+    boost::filesystem::path path( BBP_TESTDATA );
+    path /= "local/unmergedSynapses/nrn.h5";
+
+    const brion::Synapse synapseFile( path.string( ));
+    const brion::SynapseMatrix& data = synapseFile.read( 1,
+                                                brion::SYNAPSE_ALL_ATTRIBUTES );
+    std::cout << data << std::endl;
```
Comment (tribal-tec) [inline tests/synapse.cpp]: needed?

File: tests/synapse.cpp [hunk 3]
```diff
@@ -247,3 +247,34 @@ BOOST_AUTO_TEST_CASE( test_perf )
             << " synapses for " << gids.size() << " cells took: "
             << duration.total_milliseconds() << " ms." << std::endl;
 }
+
+BOOST_AUTO_TEST_CASE( read_unmerged )
```
Comment (tribal-tec) [inline tests/synapse.cpp]: add also a test for read_unmerged_invalid?

File: tests/synapse.cpp (hunk 2 above)
Comment (eile) [inline tests/synapse.cpp]: C&P. removed everywhere

File: tests/synapse.cpp (hunk 3 above)
Comment (eile) [inline tests/synapse.cpp]: Srsly? What would that test that invalid_read doesn't test already?

**Comment by eile** [comment]
retest this please

**Comment by eile** [comment]
updated - all comments addressed afaik

**Pull request merged**